libcaja-private/caja-open-with-dialog.c

```diff
@@ -249,12 +249,18 @@ add_or_find_application (CajaOpenWithDialog *dialog)
 
     if (app == NULL)
     {
-        message = g_strdup_printf (_("Could not add application to the application database: %s"), error->message);
+        if (error)
+        {
+            message = g_strdup_printf (_("Could not add application to the application database: %s"), error->message);
+            g_error_free (error);
+        }
+        else
+            message = _("unknown error");
+
         eel_show_error_dialog (_("Could not add application"),
                                message,
                                GTK_WINDOW (dialog));
         g_free (message);
-        g_error_free (error);
```

Comment:
You could use [`g_clear_error()`](https://developer.gnome.org/glib/stable/glib-Error-Reporting.html#g-clear-error) which deals with a `NULL` error:
```diff
-        g_error_free (error);
+        g_clear_error (&error);
```